hyperon_das/query_engines/query_engine_protocol.py

```diff
@@ -130,9 +130,7 @@ def query(
         raise NotImplementedError
 
     @abstractmethod
-    def custom_query(
-        self, index_id: str, query: Query, **kwargs
-    ) -> Union[Iterator, List[Dict[str, Any]]]:
+    def custom_query(self, index_id: str, query: Query, **kwargs) -> Union[Iterator, List[AtomT]]:
```

Comment:
This type hint shouldn't be `Iterator | list[AtomT]`

Reply:
Done!